README.md

```diff
@@ -48,298 +48,400 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
-  - `railId`: Rail identifier
-  - `period`: New lockup period in epochs
-  - `lockupFixed`: New fixed lockup amount
-- **Requirements**:
-  - Caller must be the rail operator
-  - For terminated rails: cannot change period or increase fixed lockup
-  - For active rails: changes restricted if client's account isn't fully funded
-  - Operator must have sufficient allowances
+-   **When to use:** An operator calls this to adjust the client's funding guarantee. This is used to set an initial `lockupFixed` for an onboarding fee, increase the `lockupPeriod` for a longer-term commitment, or decrease lockups when a deal's terms change.
+
+**Lockup Calculation and State Changes**
+
+This function recalculates the rail's total lockup requirement based on the new `period` and `lockupFixed` values. The change in the rail's individual lockup is then applied to the client's total account lockup (`Account.lockupCurrent`).
+
+-   **State Impact:** It modifies both the `Rail` struct (updating `lockupPeriod` and `lockupFixed`) and the client's `Account` struct (updating `lockupCurrent`).
+-   **Safety Checks:** The entire operation is wrapped in a modifier that settles the client's account lockup before and after the change. This ensures that any increase in lockup is only permitted if the client has sufficient available funds to cover it, maintaining the contract's solvency invariants.
+
+-   **Parameters**:
+    -   `railId`: The rail's unique identifier.
+    -   `period`: The new lockup period in epochs.
+    -   `lockupFixed`: The new fixed lockup amount.
+-   **Requirements**:
+    -   Caller must be the rail operator.
+    -   **For Terminated Rails:** The lockup period cannot be changed, and the `lockupFixed` can only be decreased.
+    -   **For Active Rails:**
+        -   Any increase to the `period` is checked against the operator's `maxLockupPeriod` allowance.
+        -   **Critical**: If the client's account is **not** fully funded (`isAccountLockupFullySettled` is false), changes are heavily restricted: the `period` cannot be changed, and `lockupFixed` can only be decreased. This prevents increasing the financial burden on an underfunded client.
 
 #### `modifyRailPayment(uint256 railId, uint256 newRate, uint256 oneTimePayment)`
 
-Modifies a rail's payment rate and / or makes a one-time payment.
+Modifies a rail's payment rate, makes an immediate one-time payment, or both.
 
-- **Parameters**:
-  - `railId`: Rail identifier
-  - `newRate`: New per-epoch payment rate
-  - `oneTimePayment`: Optional immediate payment amount
-- **Requirements**:
-  - Caller must be the rail operator
-  - For terminated rails: cannot increase rate
-  - For active rails: rate changes restricted if client's account isn't fully funded
-  - One-time payment must not exceed fixed lockup
+-   **When to use:** This is the primary function for starting a payment stream (by setting an initial `newRate`), adjusting it, or making ad-hoc payments.
+
+**Rate Change Behavior**
```

Comment:
Probably there should be a **One Time Payment Behavior** section?

Reply:
I have linked the "ONE TIME Payments" section here.